cloudera/datadog_checks/cloudera/api_client_v7.py

```diff
@@ -0,0 +1,129 @@
+from concurrent.futures import ThreadPoolExecutor
+
+import cm_client
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.cloudera.api_client import ApiClient
+from datadog_checks.cloudera.entity_status import ENTITY_STATUS
+from datadog_checks.cloudera.metrics import NATIVE_METRICS, TIMESERIES_METRICS
+
+from .common import CLUSTER_HEALTH, HOST_HEALTH
+
+
+class ApiClientV7(ApiClient):
+    def __init__(self, check, api_client):
+        super(ApiClientV7, self).__init__(check, api_client)
+
+    def collect_data(self):
+        self._collect_clusters()
+
+    def _collect_clusters(self):
+        clusters_resource_api = cm_client.ClustersResourceApi(self._api_client)
+        read_clusters_response = clusters_resource_api.read_clusters(cluster_type='any', view='full')
+        self._log.debug("Cloudera full clusters response:\n%s", read_clusters_response)
+        with ThreadPoolExecutor(max_workers=len(read_clusters_response.items) * 3) as executor:
```

Comment:
Let's add a comment explaining the choice for `max_workers`

Reply:
Will add a comment to mention that for each cluster, we are executing three tasks in parallel (line 30-32).